app/models/user/authentication.rb

```diff
@@ -0,0 +1,29 @@
+class User::Authentication < ApplicationRecord
+  include Deliverable
+  include Eventable
+
+  belongs_to :user
+
+  has_secure_token
+  has_one :session, dependent: :destroy
+
+  def expired?
+    created_at + AUTHENTICATION_VALIDITY_PERIOD < Time.now
+  end
+
+  def succeeded?
+    events.where(action: :completed).exists?
+  end
+
+  def complete
+    record(:completed)
```

Comment:
I haven't tested this, but it appears that this Event might not have a `by` User. It would be handy to have it associated to a `completed` event

Reply:
ohhh, you're right, cause we wouldn't have `Current.user` set - I can explicitly pass that in